client/my-sites/themes/theme-collections-layout.tsx

```diff
@@ -0,0 +1,76 @@
+import { memo, ReactElement } from 'react';
+import QueryThemes from 'calypso/components/data/query-themes';
+import ThemeCollection from 'calypso/components/theme-collection';
+import ThemeCollectionItem from 'calypso/components/theme-collection/theme-collection-item';
+import { ThemeBlock } from 'calypso/components/themes-list';
+import { THEME_COLLECTIONS } from 'calypso/my-sites/themes/collection-definitions';
+import { ThemesQuery, useThemeCollection } from 'calypso/my-sites/themes/use-theme-collection';
+
+interface ThemesCollectionsLayoutProps {
+	getActionLabel: ( themeId: string ) => string;
+	getOptions: ( themeId: string ) => void;
+	getScreenshotUrl: ( themeId: string ) => string;
+}
+
+function ThemeCollectionsLayout( {
+	getActionLabel,
+	getOptions,
+	getScreenshotUrl,
+}: ThemesCollectionsLayoutProps ) {
+	const ShowcaseThemeCollection = ( {
```

Comment:
I'm not convinced we should declare a component inside a component.
I presume the intent is to remove one level of props drilling, but it then becomes an antipattern and performance concern.
If we want to streamline this, we could do something like this instead:

```jsx
function ShowcaseThemeCollection( {
	collectionSlug,
	title,
	description,
	query,
	getActionLabel,
	getOptions,
	getScreenshotUrl,
} ) {
	// ...
}

function ThemeCollectionsLayout( props ) {
	return (
		<>
			<ShowcaseThemeCollection
				{ ...THEME_COLLECTIONS.premium }
				{ ...props }
			/>
		</>
	);
}
```